esp/useful_scripts/server_setup/random_password.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+# May need to be invoked with  #!/bin/bash2  on older machines.
+#
+#  Random password generator for Bash 2.x +
+#+ by Antek Sawicki <tenox@tenox.tc>,
+#+ who generously gave usage permission to the ABS Guide author.
+#
+# ==> Comments added by document author ==>
+
+
+MATRIX="0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+# ==> Password will consist of alphanumeric characters.
+LENGTH="8"
+# ==> May change 'LENGTH' for longer password.
+
+
+while [ "${n:=1}" -le "$LENGTH" ]
+# ==> Recall that := is "default substitution" operator.
+# ==> So, if 'n' has not been initialized, set it to 1.
+do
+	PASS="$PASS${MATRIX:$(($RANDOM%${#MATRIX})):1}"
```

Comment:
Also, I think Django's `SECRET_KEY` can and should be much longer than 8 characters -- iirc CSRF and other protections depend on it, and 62^8 is not huge.